lib/output.js

```diff
@@ -410,6 +410,12 @@ function OutputStream(options) {
     AST_Node.DEFMETHOD("print_to_string", function(options){
         var s = OutputStream(options);
         this.print(s);
+        if (options === undefined) {
```

Comment:
Hmm.. maybe we still do need a reset parameter, but default on restoring state afterwards. Letting the options parameter decide is more like "if you want a state reset, you must accept our way of dealing printing code".